cmake/mcuboot.cmake

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) 2020 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+# This file includes extra build system logic that is enabled when
+# CONFIG_MCUBOOT_SIGNATURE_KEY_FILE is set to a nonempty value.
+#
+# It builds signed binaries using imgtool as a post-processing step
+# after zephyr/zephyr.elf is created in the build directory.
+#
+# Since this file is brought in via include(), we do the work in a
+# function to avoid polluting the top-level scope.
+
+function(zephyr_mcuboot_tasks)
+  if(CONFIG_MCUBOOT_SIGNATURE_KEY_FILE STREQUAL "")
+    # No signature key file, no signed binaries. No warning is needed;
+    # this is the documented behavior.
+    return()
+  elseif(NOT EXISTS "${CONFIG_MCUBOOT_SIGNATURE_KEY_FILE}")
+    message(WARNING "Not signing images for MCUboot: CONFIG_MCUBOOT_SIGNATURE_KEY_FILE=\"${CONFIG_MCUBOOT_SIGNATURE_KEY_FILE}\" does not exist.")
+    return()
+  elseif(NOT WEST)
+    message(WARNING "Not signing images for MCUboot: west not found.")
```

Comment:
Should this be an error instead?

Reply:
> Should this be an error instead?

@henrikbrixandersen I think that's reasonable. The user has to take explicit action to set MCUBOOT_SIGNATURE_KEY_FILE to a non-default value so the intent is clear here.